crates/btcio/src/reader/query.rs

```diff
@@ -26,42 +26,42 @@ use crate::{
     status::{apply_status_updates, L1StatusUpdate},
 };
 
-// TODO: remove this
-pub async fn bitcoin_data_reader_task<D: Database + 'static>(
+struct ReaderContext<R: Reader> {
+    /// Bitcoin reader client
+    client: Arc<R>,
+    /// L1Event sender
+    event_tx: mpsc::Sender<L1Event>,
+    /// Config
+    config: Arc<ReaderConfig>,
+    /// Status transmitter
+    status_tx: Arc<StatusTx>,
```

Comment:
I honestly think that we should not `Arc` everything in `crates/*` since they are libs.
`Arc`ing should be done by the implementor.

Hence:
```suggestion
    client: &R,
    /// L1Event sender
    event_tx: mpsc::Sender<L1Event>,
    /// Config
    config: &ReaderConfig,
    /// Status transmitter
    status_tx: &StatusTx,
```
Then `Arc`ing it in the `bin/*` stuff.

This comment applies to all `Arc`ing in the `crates/*` changes in this PR.